common/views/components/GlobalAlertContext/GlobalAlertContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { Component, createContext, ReactNode } from 'react';
+import { HTMLString } from '@weco/common/services/prismic/types';
+const GlobalAlertContext = createContext(null);
+
+type GlobalObject = {
+  isShown: boolean | null;
+  routeRegex: string | null;
+  text: HTMLString;
+};
+
+type Props = {
+  value: GlobalObject;
+  children: ReactNode;
+};
+
+type State = {
+  isShown: boolean | null;
+  routeRegex: string | null;
+  infoBanner: boolean;
+};
+
+export class GlobalAlertContextProvider extends Component<Props, State> {
+  setInfoBanner = (value: boolean) => {
+    this.setState({
+      ...this.state,
+      infoBanner: value,
+    });
+  };
+
+  // initial context value
+  state = {
+    infoBanner: false,
+    isShown: this.props.value.isShown,
+    routeRegex: this.props.value.routeRegex,
+    text: this.props.value.text,
+    setInfoBanner: this.setInfoBanner,
+  };
+
+  render() {
+    //Passing the state object as a value prop to all children
+    return (
+      <GlobalAlertContext.Provider value={this.state}>
+        {this.props.children}
+      </GlobalAlertContext.Provider>
+    );
+  }
+}
+
+export default GlobalAlertContext;
```

Comment:
Could we call this either `GlobalAlert`, or `GlobalAlertContextProvider` so we don't sonfure it with the standard implementation of a `Context`?

Reply:
yep GlobalAlert sounds good.